---
title: "GWAS Final Report"
author: "Jake Sauter"
date: "4/29/2021"
output: 
  html_document:
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE,
                      message = FALSE,
                      comment = NA)
```

**Libraries Used**

```{r}
library(dplyr)
library(knitr)
library(ggplot2)
library(parallel)
library(magrittr)
library(patchwork)
```

## **Loading Data**

```{r, echo = FALSE}
data_path <- '../data/'
```


## **Background**

The data associated with the following analyses represents the mRNA expression levels of five different transcripts of interest (ERAP2, PEX6, FAHD1, GFM1 and MARCHF7), measured from four European or European-related populations. mRNA expression profiles were quantified through Next-Generation Sequencing (NGS) of reverse-transcribed cDNA of RNA samples dervied from Lymphoblastic Cell Lines (LCLs) generated from each individual in the study from the 1000 Genomes Project.

The phenotype data for this study was generated by the Genetic European Variation in Health and Disease (gEUVADIS). A published Nature Article [Transcriptome and genome sequencing uncovers functional variation in humans](https://www.nature.com/articles/nature12531) resulted from these data, and the data can be accessed via an [International Genome Sample Resource Data Portal](https://www.internationalgenome.org/data-portal/data-collection/geuvadis). The individuals included in the following analyses are apart of four seperate populations, **CEU** (Utah (U.S.A) residents with European ancestry), **FIN** (Finns), **GBR** (British) and **TSI** (Toscani). 

The goal of the following analysis is to reproducilby locate positions of causal single nucleotide  polymorphisms (SNPs) for the five LCL-mRNA related phenotypes aquired in the reference study. In order to accomplish this goal, a standard Genome Wide Association Study (GWAS) will be performed, in which independent hypothesis tests for association will be performed for each genotype-phenotype pair. Our aim is to identify genomic locations in which we calculate a significant association between a difference in genotype and observed phenotype, indicated by a significant ($\alpha < 0.05$) multiple testing corrected p-value for these locations. Due to the nature of the complex biological processes occuring with the information stored in genomic DNA, a significant p-value (for other than statitical reasons) may not indicate a true causual polymorphism is present at the exact location, more specifically due to linkage disequilibrium, or less formally observed correlation between closely related genomic positions.

To begin these analyses, an inspection of the reference study data will be performed, aiming to pinpoint missing, outlier, or addition trends in the data that break necessary assumptions in the analyses to follow. We will see that none of the prior mentioned abnormalities were present in the data. After data inpsection and validation, we inspect the distribution of phenotype data. The distributions of all measured RT-cDNA phenotypes appeared to be continuous and sufficiently normally distributed, allowing for analysis using linear regression. Lastly, we present the results of the five GWAS analyses by means of QQ plots (determining if the resultant p-value distribution is fairly uniform as expected) and both Global and Local Manhattan Plots. The localalized Manhattan plots are then overlayed to determine possible concordance or disconcordance 

## **Model Setup**

In order to analyze the given genetic data, we will make use of the qunantiative genetic model for a muliple regression.

**Additive Genetic Coding**

$$
X_a(A_1A_1) = -1, X_a(A_1A_2) = 0, X_a(A_2A_2) = 1,
$$

**Dominant Genetic Coding**

$$
X_d(A_1A_1) = -1, X_d(A_1A_2) = 1, X_d(A_2A_2) = -1,
$$
We will now use these encoded variables in our multiple regession model equation below

$$
Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + \epsilon
$$
where $\epsilon ~ N(0, \sigma^2)$. This set of model and encodings allow us to moel the relationship between the genotype information ($X_a$, $X_d$) to the continous phenotype ($Y$), while preserving our genetic understanding of additive and dominant allele effects. 

In the following analyses, we will use this model to perform a hypothesis test $H_0: \beta_a = 0 \cap \beta_d = 0$ (indicating that additive or dominant genetic effects were not found to be assocaited with the phenotypic difference) vs $H_a: \beta_a \neq 0 \cup \beta_d \neq 0$ (indicating that either one or both of addiditive genetic effects are associated with the observed phenotypic difference).

## **Understanding the Data**

### **Genentic Phenotype Information**

Below we see the gene information of the 5 phenotypes encoded in our file

```{r}
gene_info <-
  read.csv(file.path(data_path, 
                     'gene_info.csv'))

gene_info %>% 
  kable()
```

### **Phenotypes**

```{r}
phenotypes <- 
  read.csv(file.path(data_path, 'phenotypes.csv'), 
                       row.names = 1, header = TRUE, 
                       stringsAsFactors = FALSE) %>% 
  set_colnames(c('ERAP2', 'PEX6', 'FAHD1', 'GFM1', 'MARCH7'))
  

phenotypes %>% 
  dim()

phenotypes %>% 
  names()

phenotypes %>% 
  head() %>% 
  kable()
```


### **What Genotypes are Present?**

Below we see that we have **50,000** genotype markers for each of our **344** individuals.
Each individual is represented in a **row** in our `genotype` data frame, with each **column** corresponding the the number of minor alleles the individual has present in their DNA.

```{r}
genotypes <- 
  read.csv(file.path(data_path, 'genotypes.csv'), 
           row.names = 1, header = TRUE, 
           stringsAsFactors = FALSE)

genotypes %>% 
  dim()

genotypes %>% 
  .[1:10, 1:5] %>% 
  kable()
```

**Additive and Dominant Genetic Encoding of Genotype Information**

We can then convert this encoding to our standard $X_a$ and $X_d$ encodings (additive and dominant genetic codings) in the following way.

```{r}
Xa <- as.data.frame(genotypes) - 1
Xd <- 1 - 2*abs(Xa)
```

### **SNP Info**

```{r}
snp_info <- 
  read.csv(file.path(data_path, 
                     'SNP_info.csv'))

snp_info %>% 
  head() %>% 
  kable()
```

**Which chromosomes are represented in our samples?**

```{r}
snp_info %>% 
  .$chromosome %>% 
  unique()
```

### **Covariate Information**

Below, we see that 2 covariates were measured in this study, **Population** and **Sex**. 

```{r}
covars <- 
  read.csv(file.path(data_path, 
                     'covars.csv'), 
            row.names = 1)

covars %>% 
  head() %>% 
  kable()
```

Below we see that each covariate controlled population has over **30** samples, indicating that we should have enough data/information here in order to control for both covariates with introducing non-robust model characteristics.

```{r}
covars %>% 
  count(Population, Sex) %>% 
  kable()
```


## **Checking the Phenotype Data**

From the information above, it is clear that we are dealing with **5** phenotypes from **344** individuals, each phenotype describing the **continuous** variable of mRNA expression of the genes of interest.

### **Phenotype Histograms**

All phenotypes appear to be be normally distributed between -3 and 3. This fits our expection of Normality in GWAS 

**TODO: Why does the phenotype need to be normal in a GWAS?**

```{r}
plot_list <- vector('list', 5)
names(plot_list) <- names(phenotypes)

for (phenotype in names(phenotypes)) {
  plot_list[[phenotype]] <-
    phenotypes %>%  
      ggplot() + 
      geom_histogram(aes_string(x = phenotype), 
                     color = 'black', 
                     fill = 'skyblue') + 
    xlab('') + ylab('Frequency') + 
    ggtitle(phenotype)
}

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```

### **Checking for Odd Phenotype Values**


There does not seem to be any `NA` or outlier phenotypic values present in the data.

```{r}
phenotypes %>% 
  anyNA()
```

```{r}
phenotypes %>% 
  summary()
```

## **Check and Filter Genotype Data**

### **Data Quality Checks**

Filter genotypes:

-  Remove **individuals** with more than 10% missing data across all genotypes

-   Remove **genotypes** with more than 5% missing data across all individuals.

-   Remove **genotypes** with an MAF less than 5%

From the information below, we can see that no genotypes with an MAF of less than 5% are present in our data.

```{r}
MAFs <- 
  genotypes %>% 
  apply(2, function(x) {
    length(x[x > 0]) / length(x)
  })

MAFs %>% 
  data.frame(MAF = .) %>% 
  ggplot() + 
    geom_histogram(aes(x = MAF), 
                   color = 'black', 
                   fill = 'skyblue') + 
    xlab('Minor Allele Frequency') + 
    ylab('Frequency')

MAFs %>% 
  .[. < 0.05] %>% 
  length()
```

**TODO: Hardy-Weinburg Equilibrium?**

### **Principle Components Analysis**

```{r}
pca_result <- 
  prcomp(genotypes, 
         scale = TRUE, 
         center = TRUE)
```

```{r}
data.frame(PC1 = pca_result$x[, 1], 
           PC2 = pca_result$x[, 2], 
           Population = covars$Population, 
           Sex = covars$Sex) %>% 
ggplot() + 
  geom_point(aes(x = PC1, y = PC2, 
                 color = Population, 
                 shape = Sex)) 
```

## **Performing GWAS Without Covariates**

### **P-value Calculations**

```{r}
perform_gwas_no_covs <- function(genotypes, phenotype) { 
  n_genotypes <- ncol(genotypes)
  
  gwas_results <- 
    mclapply(seq_len(n_genotypes), 
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] + 
                                Xd[,genotype])
        
        p_val <-
          summary(model) %>% 
          .$fstatistic %>% 
          {pf(.[1], .[2], .[3], 
              lower.tail = FALSE)} 
      
        names(p_val) <- 
          colnames(genotypes)[genotype]
        
        p_val
      }, mc.cores = 7)
  
  unlist(gwas_results)
}
```

```{r}
gwas_no_covs_results <- 
  lapply(seq_len(ncol(phenotypes)), 
        function(i) {
          perform_gwas_no_covs(genotypes, 
                               phenotypes[,i])
   }) %>% 
  set_names(colnames(phenotypes))
```

### **QQ Plots**

```{r}
plot_uniform_qq <- function(p_values, pheno_name) {
  
  uniform_quantiles <- 
    qunif(ppoints(length(p_values)))
  
  p <- 
    data.frame(p_values          = sort(p_values),
               uniform_quantiles = sort(uniform_quantiles)) %>% 
    ggplot() + 
      geom_point(aes(x = -log10(uniform_quantiles),
                     y = -log10(p_values))) +
      geom_abline(intercept = 0, 
                  slope = 1, 
                  color="red", 
                  lty = 2) + 
      xlab('-log10(Uniform Quantiles)') +
      ylab('-log10(P Values)') + 
      ggtitle(pheno_name)
    
  p
}
```

```{r}
phenotype_names <- names(gwas_no_covs_results)

plot_list <- 
  lapply(seq_along(gwas_no_covs_results), 
        function(i) {
          plot_uniform_qq(gwas_no_covs_results[[i]], 
                          phenotype_names[[i]])
        })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```

### **Manhattan Plots**

```{r}

## TODO modify to use the gene info to plot the location
# of the SNP in the genome, not the number the p-value is

plot_manhattan <- function(p_values, plot_title, 
                           start = NULL, end = NULL) {
  

  ## Correlate p-values to genomic locations with snp_info
  unique_chromosomes <- 
    snp_info$chromosome %>% 
    unique()
  
  chromosome_offsets <- 
    sapply(unique_chromosomes, 
      function(chromosome) {
        offset <- 
          max(snp_info[snp_info['chromosome'] == chromosome,
                       'position'])
        names(offset) <- chromosome
        offset
    })
  
  genomic_locations <-
    snp_info[snp_info['id'] == names(p_values), ]  %>% 
    mutate(position = 
             if_else(chromosome == 1, 
                     position, 
                     position + sum(chromosome_offsets[1:chromosome-1]))) %>% 
    .$position
  
  n_genotypes <- length(p_values)
  
  df <- 
    data.frame(
      genome_location = genomic_locations, 
      pval = p_values) %>% 
    mutate(plot_pval = -log10(pval))
  
  # localizing manhattan plot to better show 
  # hits in the provided range
  if(!is.null(start) && !is.null(end)) {
    df <- df %>% 
      filter(genome_location >= start &
             genome_location <= end)
  } 
  
  
  p <- 
    df %>% 
    ggplot() + 
    geom_point(aes(x = genome_location, y = plot_pval), 
               col = 'skyblue') +
    geom_hline(aes(yintercept = -log10(0.05 / n_genotypes),
               color = 'red'), lty = 2, lwd = 1.1) +
    xlab('Genomic Location') + ylab('-log10(p value)') +
    ggtitle(plot_title) + ylim(c(0, 90)) + 
    scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) + 
    theme(legend.position = 'none', 
          axis.text.x = element_text(angle = -30)) 
  
  p
}
```


```{r}
plot_list <- 
  lapply(seq_along(gwas_no_covs_results), 
      function(i) {
        gwas_result <- gwas_no_covs_results[[i]]
        plot_name <- names(gwas_no_covs_results)[[i]]
        plot_manhattan(gwas_result, plot_name)
    })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])

```

Of the three phenotypes that we have observed significant hits for, we will take a closer look at the local regions where the hits were found to be.

```{r}
plot_manhattan(gwas_no_covs_results[['ERAP2']], 
               'ERAP2', 
               start = 9.65e7, 
               end = 9.725e7) + 
plot_manhattan(gwas_no_covs_results[['PEX6']],
               'PEX6',
               start = 4.27e7,
               end = 4.325e7) +
plot_manhattan(gwas_no_covs_results[['FAHD1']],
               'FAHD1',
               start = 1.35e6,
               end = 2.2e6)
```


## **Performing GWAS With Covariates**

### **P-value Calculations**

```{r}
perform_gwas_with_covs <- function(genotypes, phenotype) { 
  n_genotypes <- ncol(genotypes)
  
  gwas_results <- 
    mclapply(seq_len(n_genotypes), 
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] + 
                                Xd[,genotype] + 
                                covars$Sex + 
                                covars$Population)
        
        p_val <-
          summary(model) %>% 
          .$fstatistic %>% 
          {pf(.[1], .[2], .[3], 
              lower.tail = FALSE)} 
      
        names(p_val) <- 
          colnames(genotypes)[genotype]
        
        p_val
      }, mc.cores = 7)
  
  unlist(gwas_results)
}
```

```{r}
gwas_with_covs_results <- 
  lapply(seq_len(ncol(phenotypes)), 
        function(i) perform_gwas_with_covs(genotypes, phenotypes[,i])) %>% 
  set_names(colnames(phenotypes))
```


### **QQ Plots**

```{r}
phenotype_names <- names(gwas_with_covs_results)

plot_list <- 
  lapply(seq_along(gwas_with_covs_results), 
        function(i) {
          plot_uniform_qq(gwas_with_covs_results[[i]], 
                          phenotype_names[[i]])
        })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```

### **Manhattain Plots**


```{r}
plot_list <- 
  lapply(seq_along(gwas_with_covs_results), 
      function(i) {
        gwas_result <- gwas_with_covs_results[[i]]
        plot_name <- names(gwas_with_covs_results)[[i]]
        plot_manhattan(gwas_result, plot_name)
    })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])

```

Of the three phenotypes that we have observed significant hits for, we will take a closer look at the local regions where the hits were found to be. We see below that we have found the same hits with and without covariates included in our model.

```{r}
plot_manhattan(gwas_with_covs_results[['ERAP2']], 
               'ERAP2', 
               start = 9.65e7, 
               end = 9.725e7) + 
plot_manhattan(gwas_with_covs_results[['PEX6']],
               'PEX6',
               start = 4.27e7,
               end = 4.325e7) +
plot_manhattan(gwas_with_covs_results[['FAHD1']],
               'FAHD1',
               start = 1.35e6,
               end = 2.2e6)
```

## **Table of Most Significant Hits**

```{r}
data.frame(
  Phenotype = c('ERAP2', 'PEX6', 'FAHD1'), 
  '# Sig. Genotypes' = c(1, 2, 3), 
  'Top 3 Sig. Genotypes' = c('a,b,c', 
                             'a,b,c', 
                             'a,b,c')
  )%>% 
  kable()
```



## **Biological Support of Found Hits**

**Literature Review / GeneCards**
