---
title: "GWAS Final Report"
author: "Jake Sauter"
date: "4/29/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE,
                      message = FALSE,
                      comment = NA)
```

**Libraries Used**

```{r}
library(dplyr)
library(knitr)
library(ggplot2)
library(parallel)
library(magrittr)
library(patchwork)
```

## **Loading Data**

```{r, echo = FALSE}
data_path <- '../data/'
```


## **Data Background**

The data associated with the following analyses represents the mRNA expression levels of five different transcripts of interest, measured from five different populations (four of which were European), as a part of the 1000 Genomes Project. More specifically, the mRNA expression profiles were quantified through the sequencing of samples dervied from lymphoblastic cell lines (LCLs) generated for each individual in the study.

The individuals included in the following analyses are apart of four seperate populations: 

* CEU (Utah (U.S.A) residents with European ancestry)
* FIN (Finns)
* GBR (British)
* TSI (Toscani)

This data was generated by the Genetic European Variation in Health and Disease (gEUVADIS), an more information on the data is accessible via the following links: 

* [Data Portal](https://www.internationalgenome.org/data-portal/data-collection/geuvadis)
* [Associated Nature Article](https://www.nature.com/articles/nature12531)


## **Understanding the Components of the Data**


### **Genentic Phenotype Information**

Below we see the gene information of the 5 phenotypes encoded in our file

```{r}
gene_info <-
  read.csv(file.path(data_path, 
                     'gene_info.csv'))

gene_info %>% 
  kable()
```

### **Phenotypes**

```{r}
phenotypes <- 
  read.csv(file.path(data_path, 'phenotypes.csv'), 
                       row.names = 1, header = TRUE, 
                       stringsAsFactors = FALSE) %>% 
  set_colnames(c('ERAP2', 'PEX6', 'FAHD1', 'GFM1', 'MARCH7'))
  

phenotypes %>% 
  dim()

phenotypes %>% 
  names()

phenotypes %>% 
  head() %>% 
  kable()
```


### **What Genotypes are Present?**

Below we see that we have **50,000** genotype markers for each of our **344** individuals.
Each individual is represented in a **row** in our `genotype` data frame, with each **column** corresponding the the number of minor alleles the individual has present in their DNA.

```{r}
genotypes <- 
  read.csv(file.path(data_path, 'genotypes.csv'), 
           row.names = 1, header = TRUE, 
           stringsAsFactors = FALSE)

genotypes %>% 
  dim()

genotypes %>% 
  .[1:10, 1:5] %>% 
  kable()
```

**Additive and Dominant Genetic Encoding of Genotype Information**

We can then convert this encoding to our standard $X_a$ and $X_d$ encodings (additive and dominant genetic codings) in the following way.

```{r}
Xa <- as.data.frame(genotypes) - 1
Xd <- 1 - 2*abs(Xa)
```

### **SNP Info**

```{r}
snp_info <- 
  read.csv(file.path(data_path, 
                     'SNP_info.csv'))

snp_info %>% 
  head() %>% 
  kable()
```

**Which chromosomes are represented in our samples?**

```{r}
snp_info %>% 
  .$chromosome %>% 
  unique()
```

### **Covariate Information**

Below, we see that 2 covariates were measured in this study, **Population** and **Sex**. 

```{r}
covars %>% 
  head() %>% 
  kable()
```

Below we see that each covariate controlled population has over **30** samples, indicating that we should have enough data/information here in order to control for both covariates with introducing non-robust model characteristics.

```{r}
covars %>% 
  count(Population, Sex) %>% 
  kable()
```


## **Checking the Phenotype Data**

From the information above, it is clear that we are dealing with **5** phenotypes from **344** individuals, each phenotype describing the **continuous** variable of mRNA expression of the genes of interest.

### **Phenotype Histograms**

All phenotypes appear to be be normally distributed between -3 and 3. This fits our expection of Normality in GWAS 

**TODO: Why does the phenotype need to be normal in a GWAS?**

```{r}
plot_list <- vector('list', 5)
names(plot_list) <- names(phenotypes)

for (phenotype in names(phenotypes)) {
  plot_list[[phenotype]] <-
    phenotypes %>%  
      ggplot() + 
      geom_histogram(aes_string(x = phenotype), 
                     color = 'black', 
                     fill = 'skyblue') + 
    xlab('') + ylab('Frequency') + 
    ggtitle(phenotype)
}

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```

### **Checking for Odd Phenotype Values**


There does not seem to be any `NA` or outlier phenotypic values present in the data.

```{r}
phenotypes %>% 
  anyNA()
```

```{r}
phenotypes %>% 
  summary()
```

**TODO: Remove individuals missing the phenotypes of interest**?

**TODO: Show that all of these conditions are met aleady**

## **Check and Filter Genotype Data**

Filter genotypes:


-  Remove **individuals** with more than 10% missing data across all genotypes

```{r}


```


-   Remove **genotypes** with more than 5% missing data across all individuals.


-   Remove **genotypes** with an MAF less than 5%

**TODO: might actually have to filter by MAF**


-   Remove individuals that fail a test of Hardy-Weinberg equilibirum




### **Principle Components Analysis**

**test for population structure** -- clusters of individuals by genotype?

```{r}
pca_result <- 
  prcomp(genotypes, 
         scale = TRUE, 
         center = TRUE)
```

```{r}
data.frame(PC1 = pca_result$x[, 1], 
           PC2 = pca_result$x[, 2], 
           Population = covars$Population, 
           Sex = covars$Sex) %>% 
ggplot() + 
  geom_point(aes(x = PC1, y = PC2, 
                 color = Population, 
                 shape = Sex)) 
```


## **Performing GWAS**

```{r}
perform_gwas_no_covs <- function(genotypes, phenotype) { 
  n_genotypes <- ncol(genotypes)
  
  gwas_results <- 
    mclapply(seq_len(n_genotypes), 
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] + 
                                Xd[,genotype])
        
        p_val <-
          summary(model) %>% 
          .$fstatistic %>% 
          {pf(.[1], .[2], .[3], 
              lower.tail = FALSE)} %>% 
          unname()
        
        names(p_val) <- genotype
        p_val
      }, mc.cores = 10)
  
  unlist(gwas_results)
}
```

```{r}
gwas_no_cov_results <- 
  lapply(seq_len(ncol(phenotypes)), 
        function(i) perform_gwas_no_covs(genotypes, phenotypes[,i]))
```


## **GWAS Diagnositcs**

### **P-value Distributions**

```{r}
plot_list <- vector('list', 5)

for (i in seq_along(plot_list)) {
  plot_list[[i]] <- 
    all_gwas_results[[i]] %>% 
      data.frame(p_values = .) %>% 
      ggplot() + 
        geom_histogram(aes(x = p_values)) + 
        xlab('P Value') + ylab('Frequency')
}

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```


### **QQ Plots**

```{r}
plot_uniform_qq <- function(p_values) {
  
  uniform_quantiles <- 
    qunif(ppoints(length(p_values)))
  
  p <- 
    data.frame(p_values          = sort(p_values),
               uniform_quantiles = sort(uniform_quantiles)) %>% 
    ggplot() + 
      geom_point(aes(x = -log10(uniform_quantiles),
                     y = -log10(p_values))) +
      geom_abline(intercept = 0, 
                  slope = 1, 
                  color="red", 
                  lty = 2) + 
      xlab('-log10(Uniform Quantiles)') +
      ylab('-log10(P Values)')
    
  p
}
```

```{r}
plot_list <- lapply(all_gwas_results, 
                    function(gwas_res) 
                      plot_uniform_qq(gwas_res))

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```




For **each** phenotype (5 present here),

-   Perform an association analysis considering the associates of **each** marker **one at a time**
-   Check the **QQ Plots** for each individual GWAS analysis to ensure that the p-value distribution is close to uniform, meaning we have captured all necessary covariates, as well as have **a few** very significant p-values

### **Comparing Hits Between GWAS Analyses**

## **Final Analysis Presentation**

### **List all steps and methods**

### **Manhattan Plots**

### **QQ Plots**

### **Presenting Hits**

### **Biological Support of Found Hits**

**Literature Review / GeneCards**
