---
title: "GWAS Final Report"
author: "Jake Sauter"
date: "5/14/2021"
output: 
  pdf_document
header-includes:
  - \usepackage{titling}
  - \setlength{\droptitle}{-5em} 
  - \setlength{\belowcaptionskip}{1pt}
  - \setlength{\abovecaptionskip}{-2pt}
fontsize: 10.5pt
geometry: margin=0.85in
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE,
                      message = FALSE,
                      comment = NA, 
                      echo = FALSE)
```

```{r, include = FALSE}
library(dplyr)
library(knitr)
library(ggplot2)
library(parallel)
library(magrittr)
library(patchwork)
library(kableExtra)
```

```{r, include = FALSE}
data_path <- '../data/'

gene_info <-
  read.csv(file.path(data_path, 
                     'gene_info.csv'))

phenotypes <- 
  read.csv(file.path(data_path, 'phenotypes.csv'), 
                       row.names = 1, header = TRUE, 
                       stringsAsFactors = FALSE) %>% 
  set_colnames(c('ERAP2', 'PEX6', 'FAHD1', 'GFM1', 'MARCH7'))
  
genotypes <- 
  read.csv(file.path(data_path, 'genotypes.csv'), 
           row.names = 1, header = TRUE, 
           stringsAsFactors = FALSE)

covars <- 
  read.csv(file.path(data_path, 
                     'covars.csv'), 
            row.names = 1)

snp_info <- 
  read.csv(file.path(data_path, 
                     'SNP_info.csv'))
```


# **Background**

The data associated with the following analyses represents the mRNA expression levels of five different transcripts of interest (ERAP2, PEX6, FAHD1, GFM1 and MARCHF7), measured from four European or European-related populations. mRNA expression profiles were quantified through Next-Generation Sequencing (NGS) of reverse-transcribed cDNA of RNA samples dervied from Lymphoblastic Cell Lines (LCLs) generated from each individual in the study, originating from the 1000 Genomes Project.

The phenotype data for this study was generated by the Genetic European Variation in Health and Disease (gEUVADIS). A published Nature Article ([Transcriptome and genome sequencing uncovers functional variation in humans](https://www.nature.com/articles/nature12531)) resulted from these data, and the data can be accessed via an [International Genome Sample Resource Data Portal](https://www.internationalgenome.org/data-portal/data-collection/geuvadis). The individuals included in the following analyses are apart of four seperate populations, **CEU** (Utah, U.S.A residents with European ancestry), **FIN** (Finns), **GBR** (British) and **TSI** (Toscani, Italy). 

The goal of the following analysis is to reproducilby locate positions of causal single nucleotide  polymorphisms (SNPs) for the five LCL-mRNA related phenotypes aquired in the reference study. In order to accomplish this goal, a standard Genome Wide Association Study (GWAS) will be applied, in which independent hypothesis tests for association will be performed for each genotype-phenotype pair. Our aim is to identify genomic locations in which we calculate a significant association between a difference in genotype and observed phenotype, indicated by a significant ($\alpha < 0.05$) multiple testing corrected (Bonferroni) p-value for these locations. Due to the nature of the complex biological processes involved in the process of transcription, a significant p-value (for other than statitical reasons) may not indicate a true causual polymorphism is present at the exact location, more specifically due to linkage disequilibrium, or less formally, observed correlation between closely co-located genomic positions.

To begin these analyses, an inspection of the reference study data will be performed, aiming to pinpoint missing, outlier, or additional trends in the data that may break necessary assumptions in the analyses to follow. After data inpsection and validation, we will inspect the distribution of phenotype data. The distributions of all measured RT-cDNA phenotypes appeared to be continuous with no outlier values. Lastly, we present the results of the five GWAS analyses by means of QQ plots (determining if the resultant p-value distribution is fairly uniform as expected) and both global and localized Manhattan Plots. The localalized Manhattan plots are then compared to determine possible concordance or disconcordance of significant genomic locations between the five genotypes analyzed in this study.

# **Model Setup**

In order to analyze the given genetic data, we will make use of the qunantiative genetic model for muliple regression. Both an additive ($X_a(A_1A_1) = -1, X_a(A_1A_2) = 0, X_a(A_2A_2) = 1$ and dominant ($X_d(A_1A_1) = -1, X_d(A_1A_2) = 1, X_d(A_2A_2) = -1$) genetic coding will be utitlized in our multiple regession model equation ($Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + \epsilon$)
where $\epsilon \sim N(0, \sigma^2)$. This set of model and encodings allow us to capture the relationship between the genotype information ($X_a$, $X_d$) and the continous phenotype ($Y$), while preserving our genetic understanding of additive and dominant allele effects. 

In the following analyses, we will use this model to perform a hypothesis test with a null hypothesis $H_0: \beta_a = 0 \cap \beta_d = 0$ (indicating that additive or dominant genetic effects were not found to be assocaited with the phenotypic difference) vs our alternative hypothesis $H_a: \beta_a \neq 0 \cup \beta_d \neq 0$ (indicating that either one or both of addiditive and dominant genetic effects are associated with the observed phenotypic difference).

# **Understanding the Data**

## **Phenotype Data**

The phenotype data for this study has been provided in a relatively common and expected format, with one row per individual measured, and one column for each of the measured phenotypes. A sample selection of the phenotype data for six individuals is presented in Table 1.

```{r}
phenotypes %>% 
  head() %>% 
  kable(caption = 'Example rows of the phenotype data for six individuals in this study.') %>% 
  kable_styling(font_size = 10, 
                latex_options = c('hold_position'))
```


## **Genotype Data**

A small sample of the $50,000$ genotype markers for each of our $344$ individuals in the study is presented in Table 2. Each individual is represented in a row in our genotype data frame, with each column corresponding the the number of minor alleles the individual has present in their DNA.

```{r}
genotypes %>% 
  .[1:3, 1:5] %>% 
  kable(caption = 'Example rows and columns of the genotype data for five genotypes and three individuals. ') %>% 
  kable_styling(font_size = 10, 
                latex_options = c('hold_position'))
```

We can now convert this encoding to our standard $X_a$ and $X_d$ encodings (additive and dominant genetic codings) with the following equations: $X_a = \text{genotypes} - 1$, $X_d = 1 - 2 \cdot |X_a|$. 

```{r, include = FALSE}
Xa <- as.data.frame(genotypes) - 1
Xd <- 1 - 2*abs(Xa)
```


## **Covariate Information**

Covariates of population and sex were measured in this study. In this report we will model the genotype-phenotype relationship both with and without including these covariates.

An important aspect of including covariates while modelling is ensuring a stable and informative model. Practially, this requires a sufficient sample size per covariate-defined population. Table 3 shows that each covariate controlled population has over $30$ samples, indicating that we should have a sufficiently representitive samples in order to control for both covariates without introducing non-robust model characteristics.

```{r}
covars %>%
  count(Population, Sex) %>%
  rename('Num. Observed' = n) %>%
  kable(caption = 'Count of individuals in the study divided by the covariates of population and sex.') %>%
  kable_styling(font_size = 10, 
                latex_options = c('hold_position'))
```


## **Phenotype Data Quality Control**

From the information in previous sections, it is clear that we will be working with $5$ phenotypes from $344$ individuals, each phenotype describing the continuous variable of mRNA expression of the genes of interest. No missing values were identified in the phenotype data, and all phenotypes appear to be be normally distributed in the range ($-3, 3)$, shown in Figure 1.

```{r, fig.height = 2.5, fig.cap = 'Histograms of phenotype expression levels for all populations.'}
plot_list <- vector('list', 5)
names(plot_list) <- names(phenotypes)

for (phenotype in names(phenotypes)) {
  plot_list[[phenotype]] <-
    phenotypes %>%
      ggplot() +
      geom_histogram(aes_string(x = phenotype),
                     color = 'black',
                     fill = 'skyblue') +
    xlab('') + ylab('Frequency') +
    ggtitle(phenotype) 
}

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]]) 
# + 
  # plot_annotation(caption = 'Figure 1: Histograms of phenotype expression levels for all populations.')
```

## **Genotype Data Quality Control**

Common techniques used to filter GWAS study data usually include removing **individuals** with more than 10% missing data across all genotypes, removing **genotypes** with more than 5% missing data across all individuals, and removing **genotypes** with an MAF less than 5%. No missing values were identified in the gentoype data, thus the first two points are satisfied. From the histogram in Figure 2, we can see that no genotypes with an MAF of less than 5% are present in our data.

```{r, fig.height = 1, fig.cap = 'Histogram of minumum allele frequecy (MAF) for all combined populations.'}
MAFs <-
  genotypes %>%
  apply(2, function(x) {
    length(x[x > 0]) / length(x)
  })

MAFs %>%
  data.frame(MAF = .) %>%
  ggplot() +
    geom_histogram(aes(x = MAF),
                   color = 'black',
                   fill = 'skyblue') +
    geom_vline(aes(xintercept = 0.05,
                   color = 'red_dotted_line'),
               lty = 2)  +
    xlab('Minor Allele Frequency') +
    ylab('Frequency') +
    scale_color_manual(values = c('red_dotted_line' = 'red'),
                       labels = c('MAF = 5%')) +
  theme(legend.title = element_blank()) 
# +
  # labs(caption = 'Figure 2: Histogram of minumum allele frequecy (MAF) for all combined populations.')
```

## **Principle Components Analysis**

In order to understand if any other population structure exists in the data other than the captured population identifiers in the covariates, we now perform Principle Components Analysis (PCA) on the genotype data of each population seperately, and plot the population-seperated data by the two most discriptivive principle components.

Figure 2 shows that **CEU** and **TSI** populations are fairly uniform, and slight seperation is seen in both **FIN** and **GBR**. Both of these observed sub-populations do not seem to be extreme or due to the sex of the individual (indicated by the color of the points).

```{r}
unique_populations <-
  covars$Population %>%
  unique()

pca_results <- vector('list', 4)

for (i in seq_along(unique_populations)) {
  population <- unique_populations[i]

  pop_data <-
    genotypes %>%
    .[covars$Population == population, ]

  ## Remove constant / zero columns as to not cause a
  # problem with the scale. parameter below
  keep_cols <-
    pop_data %>%
    apply(2, function(x) {
      if (near(var(x), 0)) return(FALSE)
      TRUE
    })

  pop_data <-
    pop_data[, keep_cols]

  pca_results[[i]]<-
    prcomp(pop_data,
           center = TRUE,
           scale. = TRUE)
}
```

```{r, fig.cap = 'Top two principle components for the genotype information for each population.', fig.height = 3.5}
plot_list <- vector('list', 5)
names(plot_list) <- unique_populations

for (i in seq_along(unique_populations)) {
  population <- unique_populations[i]

  plot_list[[i]] <-
    data.frame(PC1 = pca_results[[i]]$x[, 1],
               PC2 = pca_results[[i]]$x[, 2],
               Sex = covars[covars$Population == population, 'Sex']) %>%
    ggplot() +
      geom_point(aes(x = PC1, y = PC2,
                     color = Sex)) +
    ggtitle(population) +
    theme(legend.position = 'none')
}

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]]) 
# + 
    # plot_annotation(caption = 'Figure 3: Top two principle components for the genotype information for each population.')
```


# **Performing GWAS Without Covariates**

We will first perfrom the five GWAS analyses without any included covariates, thus the linear models we will be fitting are of the form $Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + \epsilon$
where $\epsilon ~ N(0, \sigma^2)$. The models were fit using the `lm()` function in the R programming language, followed by calculation of the p-value through the models F-statistic on a distribution with the appropriate degrees of freedom.


```{r}
perform_gwas_no_covs <- function(genotypes, phenotype) {
  n_genotypes <- ncol(genotypes)

  gwas_results <-
    mclapply(seq_len(n_genotypes),
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] +
                                Xd[,genotype])

        p_val <-
          summary(model) %>%
          .$fstatistic %>%
          {pf(.[1], .[2], .[3],
              lower.tail = FALSE)}

        names(p_val) <-
          colnames(genotypes)[genotype]

        p_val
      }, mc.cores = 7)

  unlist(gwas_results)
}
```

```{r}
gwas_no_covs_results <-
  lapply(seq_len(ncol(phenotypes)),
        function(i) {
          perform_gwas_no_covs(genotypes,
                               phenotypes[,i])
   }) %>%
  set_names(colnames(phenotypes))
```

## **QQ Plots**

In order to observe if the resultant ditribution of p-values for each GWAS is approximately normal, as we would expect under $H_0$, we produce a Quantile-Quantile plot in Figure 4 comparing the quantiles of the p-value distribution with that of a uniform distribution.


```{r}
plot_uniform_qq <- function(p_values, pheno_name) {

  uniform_quantiles <-
    qunif(ppoints(length(p_values)))

  p <-
    data.frame(p_values          = sort(p_values),
               uniform_quantiles = sort(uniform_quantiles)) %>%
    ggplot() +
      geom_point(aes(x = -log10(uniform_quantiles),
                     y = -log10(p_values))) +
      geom_abline(intercept = 0,
                  slope = 1,
                  color="red",
                  lty = 2) +
      xlab('-log10(Uniform Quantiles)') +
      ylab('-log10(P Values)') +
      ggtitle(pheno_name)

  p
}
```

```{r, fig.height = 2.5, fig.cap = 'Quantile-Quantile plots of resultant p-values vs. Uniform quanitles for each of the five GWAS analyses performed.'}
phenotype_names <- names(gwas_no_covs_results)

plot_list <-
  lapply(seq_along(gwas_no_covs_results),
        function(i) {
          plot_uniform_qq(gwas_no_covs_results[[i]],
                          phenotype_names[[i]])
        })

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]]) 
# + 
    # plot_annotation(caption = 'Figure 4: Quantile-Quantile plots of resultant p-values vs. Uniform quanitles for each of the five GWAS analyses performed.')
```


## **Manhattan Plots**

To illistate the location and significance of the detected hits in the analyses perform, we plot the $-log_{10}(pvalue)$ of each genotype at the position where the genotype is found in the human genome in Figure 5.


```{r}
plot_manhattan <- function(p_values, plot_title,
                           start = NULL, end = NULL) {


  ## Correlate p-values to genomic locations with snp_info
  unique_chromosomes <-
    snp_info$chromosome %>%
    unique()

  chromosome_offsets <-
    sapply(unique_chromosomes,
      function(chromosome) {
        offset <-
          max(snp_info[snp_info['chromosome'] == chromosome,
                       'position'])
        names(offset) <- chromosome
        offset
    })

  genomic_locations <-
    snp_info[snp_info['id'] == names(p_values), ]  %>%
    mutate(position =
             if_else(chromosome == 1,
                     position,
                     position + sum(chromosome_offsets[1:chromosome-1]))) %>%
    .$position

  n_genotypes <- length(p_values)

  df <-
    data.frame(
      genome_location = genomic_locations,
      pval = p_values) %>%
    mutate(plot_pval = -log10(pval))

  # localizing manhattan plot to better show
  # hits in the provided range
  if(!is.null(start) && !is.null(end)) {
    df <- df %>%
      filter(genome_location >= start &
             genome_location <= end)
  }


  p <-
    df %>%
    ggplot() +
    geom_point(aes(x = genome_location, y = plot_pval),
               col = 'skyblue', shape = 21) +
    geom_hline(aes(yintercept = -log10(0.05 / n_genotypes),
               color = 'red'), lty = 2, lwd = 1.1) +
    xlab('Genomic Location') + ylab('-log10(p value)') +
    ggtitle(plot_title) + ylim(c(0, 90)) +
    scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
    theme(legend.position = 'none',
          axis.text.x = element_text(angle = -30))

  p
}
```


```{r, fig.height = 3, fig.cap = 'Manhattan plots showing global genomic view of most statistically significant associated genotypes for each phenotype. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.'}
plot_list <-
  lapply(seq_along(gwas_no_covs_results),
      function(i) {
        gwas_result <- gwas_no_covs_results[[i]]
        plot_name <- names(gwas_no_covs_results)[[i]]
        plot_manhattan(gwas_result, plot_name)
    })

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]]) 
# + 
    # plot_annotation(caption = 'Figure 5: Manhattan plots showing global genomic view of most statistically significant associated genotypes for each phenotype. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.')
```

Of the three phenotypes that we have observed significant hits for, we show in Figure 6 a closer look at the local regions where the hits were found to be.

```{r, fig.height = 2, fig.cap='Localized Manhattan plots for each phenotype that was observed in Figure 5 to return significant p-values. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.'}
plot_manhattan(gwas_no_covs_results[['ERAP2']],
               'ERAP2',
               start = 9.65e7,
               end = 9.725e7) +
plot_manhattan(gwas_no_covs_results[['PEX6']],
               'PEX6',
               start = 4.27e7,
               end = 4.325e7) +
plot_manhattan(gwas_no_covs_results[['FAHD1']],
               'FAHD1',
               start = 1.35e6,
               end = 2.2e6) 
# + 
    # plot_annotation(caption = 'Figure 6: Localized Manhattan plots for each phenotype that was observed in Figure 5 to return significant p-values. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.')
```


# **Performing GWAS With Covariates**

We now perfrom the same five GWAS analyses with the inclusion of the collected population and sex covariates, thus the linear models we will be fitting are of the form $Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + X_{Sex} \beta_{Sex} + X_{Population} \beta_{Population} + \epsilon$
where $\epsilon \sim N(0, \sigma^2)$. The models were fit similarly using the `lm()` function in the R programming language, followed by calculation of the p-value through the models F-statistic on a distribution with the appropriate degrees of freedom.

```{r}
perform_gwas_with_covs <- function(genotypes, phenotype) {
  n_genotypes <- ncol(genotypes)

  gwas_results <-
    mclapply(seq_len(n_genotypes),
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] +
                                Xd[,genotype] +
                                covars$Sex +
                                covars$Population)

        p_val <-
          summary(model) %>%
          .$fstatistic %>%
          {pf(.[1], .[2], .[3],
              lower.tail = FALSE)}

        names(p_val) <-
          colnames(genotypes)[genotype]

        p_val
      }, mc.cores = 7)

  unlist(gwas_results)
}
```

```{r}
gwas_with_covs_results <-
  lapply(seq_len(ncol(phenotypes)),
        function(i) perform_gwas_with_covs(genotypes, phenotypes[,i])) %>%
  set_names(colnames(phenotypes))
```


## **QQ Plots**

In order to observe if the resultant ditribution of p-values for each GWAS is approximately normal, as we would expect under $H_0$, Figure 7 shows a Quantile-Quantile plot comparing the quantiles of the p-value distribution with that of a uniform distribution.

```{r, fig.height = 2.5, fig.cap = 'Quantile-Quantile plots of resultant p-values vs. Uniform quanitles for each of the five GWAS analyses (with covariates included) performed.'}
phenotype_names <- names(gwas_with_covs_results)

plot_list <-
  lapply(seq_along(gwas_with_covs_results),
        function(i) {
          plot_uniform_qq(gwas_with_covs_results[[i]],
                          phenotype_names[[i]])
        })

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]]) 
# + 
    # plot_annotation(caption = 'Figure 7: Quantile-Quantile plots of resultant p-values vs. Uniform quanitles for each of the five GWAS analyses (with covariates included) performed.')
```


## **Manhattain Plots**

To illistate the location and significance of the detected hits in the analyses perform, Figure 8 plots the $-log_{10}(pvalue)$ of each genotype at the position where the genotype is found in the human genome.

```{r, fig.height = 3, fig.cap='Manhattan plots showing global genomic view of most statistically significant associated genotypes for each phenotype, with the analyses now including covariates. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.'}
plot_list <-
  lapply(seq_along(gwas_with_covs_results),
      function(i) {
        gwas_result <- gwas_with_covs_results[[i]]
        plot_name <- names(gwas_with_covs_results)[[i]]
        plot_manhattan(gwas_result, plot_name)
    })

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])  
# + 
    # plot_annotation(caption = 'Figure 8: Manhattan plots showing global genomic view of most statistically significant associated genotypes for each phenotype, with the analyses now including covariates. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.')
```

Of the three phenotypes that we have observed significant hits for, Figure 9 takes a closer look at the local regions where the hits were found to be. We see below that we have found the same hits with and without covariates included in our model.

```{r, fig.height = 2, fig.cap = 'Localized Manhattan plots for each phenotype that was observed in Figure 8 to return significant p-values. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.'}
plot_manhattan(gwas_with_covs_results[['ERAP2']],
               'ERAP2',
               start = 9.65e7,
               end = 9.725e7) +
plot_manhattan(gwas_with_covs_results[['PEX6']],
               'PEX6',
               start = 4.27e7,
               end = 4.325e7) +
plot_manhattan(gwas_with_covs_results[['FAHD1']],
               'FAHD1',
               start = 1.35e6,
               end = 2.2e6) 
# + 
    # plot_annotation(caption = 'Figure 9: Localized Manhattan plots for each phenotype that was observed in Figure 8 to return significant p-values. The horizontal red-dotted line indicates the Bonferroni corrected significance threshold.')
```

\newpage

# **Table of Most Significant Hits**

```{r}
top_sig_hits_no_covs <- vector('list', 5)
top_sig_hits_with_covs <- vector('list', 5)
num_sig_no_covs <- vector('numeric', 5)
num_sig_with_covs <- vector('numeric', 5)

names(top_sig_hits_no_covs) <- phenotype_names
names(top_sig_hits_no_covs) <- phenotype_names
names(num_sig_no_covs) <- phenotype_names
names(num_sig_with_covs) <- phenotype_names

for (phenotype in phenotype_names) {
  num_sig_no_covs[phenotype] <-
    gwas_no_covs_results[[phenotype]] %>%
    {. * length(.)} %>% {. < 0.05} %>% which() %>% length()

  num_sig_with_covs[phenotype] <-
    gwas_with_covs_results[[phenotype]] %>%
    {. * length(.)} %>% {. < 0.05} %>% which() %>% length()

  top_sig_hits_no_covs[[phenotype]] <-
    gwas_no_covs_results[[phenotype]] %>%
    sort() %>% head(3) %>% names() %>% paste0(collapse = ',')

  top_sig_hits_with_covs[[phenotype]] <-
    gwas_with_covs_results[[phenotype]] %>%
    sort() %>% head(3) %>% names() %>% paste0(collapse = ',')
}
```


```{r, eval = FALSE}
data.frame(
  Phenotype = c('ERAP2', 'PEX6', 'FAHD1'),
  'Num Sig. without covs' = c(num_sig_no_covs['ERAP2'],
                                    num_sig_no_covs['PEX6'],
                                    num_sig_no_covs['FAHD1']),
  'Num Sig. with covs' =  c(num_sig_with_covs['ERAP2'],
                                    num_sig_with_covs['PEX6'],
                                    num_sig_with_covs['FAHD1']),
  'Top 3 Sig. without covs' = c(top_sig_hits_no_covs[['ERAP2']],
                                      top_sig_hits_no_covs[['PEX6']],
                                      top_sig_hits_no_covs[['FAHD1']]),
  'Top 3 Sig. with' = c(top_sig_hits_with_covs[['ERAP2']],
                                     top_sig_hits_with_covs[['PEX6']],
                                     top_sig_hits_with_covs[['FAHD1']])
  )
```


Table 4: Count of most significant genotypes for each phenotype that returned significant results, along with the three most significant genotypes for each phenotype.

Analysis | ERAP2 # of Significant Genotypes | ERAP2 3 Most Significant Genotypes | PEX6 # of Significant Genotypes | PEX6 3 Most Significant Genotypes | FAHD1 # of Significant Genotypes | FAHD1 3 Most Significant Genotypes
--- | --- | --- | --- | --- | --- | ---
Without Covariates | 73 | rs7726445, rs7731592, rs2161548 | 29 | rs1129187, rs10948061, rs9471985 | 90 | rs11644748, rs140254902, rs9652776
--- | --- | --- | --- | --- | --- | ---
With Covariates | 75 | rs7726445, rs7731592, rs2161548 | 26 | rs1129187, rs10948061, rs9471985 | 89 | rs11644748, rs140254902, rs9652776


# **Biological Support of Found Hits**

Observing the localized Manhattan plots in Figure 6 and Figure 9, we can see that the associated gentoypes of each of the measured phenotypes were not similar between phenotypes. We also observe that many genotypes in the same region are implicated, informing us that we should also search for the relevance of nearby SNPs to our most signifcant. In order to determine the biological relevance and implications of these genotypes, we searched published research online for previously investigated causal polymorphisms that lie in the same region of the genome as our identified associated SNPs. 

The SNP rs2927608 has been previously indicated as a causal genotype for ERAP2 expression **[1]**. This SNP is located in position 96916728 of the human genome, within the region of many of our significant genotypes for ERAP2. This genotype has been found to be associated with inflammatory bowel disease through an increased gene expression of ERAP2. This genotype accounted for nearly 75% of the total variability of ERAP2 expression in blood in this study, demonstrating its significance.

The SNP rs10948059 was previously found to be a possible causal genotype for PEX6, and is located at position 42960723 in the genome. This SNP also corresponds to the genomic locus where many of our significant genotypes were found in our GWAS analysis. rs10948059 has been associated with prostate cancer from previous association studies **[2]**, so much so that the cited study implies that variants associated with prostate cancer can be identified through PEX6 gene expression changes.

The SNP rs1065656 has been previously implicated as a possible causal genotype for FAHD1, and is located at position 1788835 in the genome. We have again found many significant genotypes at this location in our GWAS analysis. One study found have demonstrated that rs1065656 is associated with circulating IGF‐I and IGFBP‐3 concentration, which have been identied as risk factors in human diseases including cardiovascular diseases, diabetes, and cancer **[3]**.


# **References**

**[1]** Di Narzo, A., Peters, L., Argmann, C., Stojmirovic, A., Perrigoue, J., Li, K., . . . Hao, K. (2016, June23). Blood and Intestine Eqtls from AN Anti-TNF-Resistant Crohn’s DISEASE Cohort inform IBDgenetic association loci. Retrieved May 04, 2021, from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4931595

**[2]** Lee, J., Ryu, J., & Lee, C. (2016, December). Strong cis-acting expression quantitative trait loci for thegenes encoding snhg5 and pex6. Retrieved May 04, 2021, from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5207599

**[3]** Teumer, A., Qi, Q., Nethander, M., Aschard, H., Bandinelli, S., Beekman, M., . . . Kaplan, R. (2016,October). Genomewide meta-analysis IDENTIFIES loci associated WITH IGF-I and IGFBP-3 levelswith impact on age-related traits. Retrieved May 04, 2021, from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5013013
