---
title: "GWAS Final Report"
author: "Jake Sauter"
date: "5/14/2021"
output: 
  pdf_document
header-includes:
  - \usepackage{titling}
  - \setlength{\droptitle}{-5em} 
fontsize: 10.5pt
geometry: margin=0.85in
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE,
                      message = FALSE,
                      comment = NA, 
                      echo = FALSE)
```

```{r, include = FALSE}
library(dplyr)
library(knitr)
library(ggplot2)
library(parallel)
library(magrittr)
library(patchwork)
```

```{r, include = FALSE}
data_path <- '../data/'

gene_info <-
  read.csv(file.path(data_path, 
                     'gene_info.csv'))

phenotypes <- 
  read.csv(file.path(data_path, 'phenotypes.csv'), 
                       row.names = 1, header = TRUE, 
                       stringsAsFactors = FALSE) %>% 
  set_colnames(c('ERAP2', 'PEX6', 'FAHD1', 'GFM1', 'MARCH7'))
  
genotypes <- 
  read.csv(file.path(data_path, 'genotypes.csv'), 
           row.names = 1, header = TRUE, 
           stringsAsFactors = FALSE)

covars <- 
  read.csv(file.path(data_path, 
                     'covars.csv'), 
            row.names = 1)

snp_info <- 
  read.csv(file.path(data_path, 
                     'SNP_info.csv'))
```


## **Background**

The data associated with the following analyses represents the mRNA expression levels of five different transcripts of interest (ERAP2, PEX6, FAHD1, GFM1 and MARCHF7), measured from four European or European-related populations. mRNA expression profiles were quantified through Next-Generation Sequencing (NGS) of reverse-transcribed cDNA of RNA samples dervied from Lymphoblastic Cell Lines (LCLs) generated from each individual in the study, originating from the 1000 Genomes Project.

The phenotype data for this study was generated by the Genetic European Variation in Health and Disease (gEUVADIS). A published Nature Article ([Transcriptome and genome sequencing uncovers functional variation in humans](https://www.nature.com/articles/nature12531)) resulted from these data, and the data can be accessed via an [International Genome Sample Resource Data Portal](https://www.internationalgenome.org/data-portal/data-collection/geuvadis). The individuals included in the following analyses are apart of four seperate populations, **CEU** (Utah (U.S.A) residents with European ancestry), **FIN** (Finns), **GBR** (British) and **TSI** (Toscani). 

The goal of the following analysis is to reproducilby locate positions of causal single nucleotide  polymorphisms (SNPs) for the five LCL-mRNA related phenotypes aquired in the reference study. In order to accomplish this goal, a standard Genome Wide Association Study (GWAS) will be performed, in which independent hypothesis tests for association will be performed for each genotype-phenotype pair. Our aim is to identify genomic locations in which we calculate a significant association between a difference in genotype and observed phenotype, indicated by a significant ($\alpha < 0.05$) multiple testing corrected p-value for these locations. Due to the nature of the complex biological processes occuring with the information stored in genomic DNA, a significant p-value (for other than statitical reasons) may not indicate a true causual polymorphism is present at the exact location, more specifically due to linkage disequilibrium, or less formally, observed correlation between closely co-located genomic positions.

To begin these analyses, an inspection of the reference study data will be performed, aiming to pinpoint missing, outlier, or additional trends in the data that may break necessary assumptions in the analyses to follow. After data inpsection and validation, we will inspect the distribution of phenotype data. The distributions of all measured RT-cDNA phenotypes appeared to be continuous and sufficiently normally distributed, allowing for analysis using linear regression. Lastly, we present the results of the five GWAS analyses by means of QQ plots (determining if the resultant p-value distribution is fairly uniform as expected) and both global and localized Manhattan Plots. The localalized Manhattan plots are then overlayed to determine possible concordance or disconcordance of significant genomic locations between the five genotypes analyzed in this study.

## **Model Setup**

In order to analyze the given genetic data, we will make use of the qunantiative genetic model for a muliple regression. Both an additive ($X_a(A_1A_1) = -1, X_a(A_1A_2) = 0, X_a(A_2A_2) = 1$ and dominant ($X_d(A_1A_1) = -1, X_d(A_1A_2) = 1, X_d(A_2A_2) = -1$ genetic coding will be utitlized in our multiple regession model equation ($Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + \epsilon$)
where $\epsilon ~ N(0, \sigma^2)$. This set of model and encodings allow us to capture the relationship between the genotype information ($X_a$, $X_d$) and the continous phenotype ($Y$), while preserving our genetic understanding of additive and dominant allele effects. 

In the following analyses, we will use this model to perform a hypothesis test with a null hypothesis $H_0: \beta_a = 0 \cap \beta_d = 0$ (indicating that additive or dominant genetic effects were not found to be assocaited with the phenotypic difference) vs our alternative hypothesis $H_a: \beta_a \neq 0 \cup \beta_d \neq 0$ (indicating that either one or both of addiditive genetic effects are associated with the observed phenotypic difference).

## **Understanding the Data**

### **Phenotype Data**

The phenotype data for this study has been provided in a relatively common and expected format, with one row per individual meansured, and one column for each of the measured phenotypes. A sample selection of the phenotype data for six individuals is shown below.

```{r}
phenotypes %>% 
  head() %>% 
  kable()
```


### **Genotype Data**

Below we see that we have **50,000** genotype markers for each of our **344** individuals. Each individual is represented in a **row** in our `genotype` data frame, with each **column** corresponding the the number of minor alleles the individual has present in their DNA.

```{r}
genotypes %>% 
  .[1:3, 1:5] %>% 
  kable()
```

We can now convert this encoding to our standard $X_a$ and $X_d$ encodings (additive and dominant genetic codings) in the following way.

```{r, include = TRUE, echo = TRUE}
Xa <- as.data.frame(genotypes) - 1
Xd <- 1 - 2*abs(Xa)
```


### **Covariate Information**

Covariates of **Population** and **Sex** were measured in this study. In this report we will model the genotype-phenotype relationship both with and without including these covariates.

An important aspect of including covariates while modelling is ensuring a stable and informative model. Practially, this required a sufficient sample size per covariate-defined population. Below we see that each covariate controlled population has over **30** samples, indicating that we should have enough data/information here in order to control for both covariates with introducing non-robust model characteristics.

```{r}
covars %>% 
  count(Population, Sex) %>% 
  rename('Num. Observed' = n) %>% 
  kable()
```


## **Phenotype Data Quality Control**

From the information above, it is clear that we will be working with **5** phenotypes from **344** individuals, each phenotype describing the continuous variable of mRNA expression of the genes of interest. No missing values were identified in the phenotype data, and all phenotypes appear to be be normally distributed between -3 and 3. This fits our expection of Normality in GWAS. 

```{r, fig.height = 3.5}
plot_list <- vector('list', 5)
names(plot_list) <- names(phenotypes)

for (phenotype in names(phenotypes)) {
  plot_list[[phenotype]] <-
    phenotypes %>%  
      ggplot() + 
      geom_histogram(aes_string(x = phenotype), 
                     color = 'black', 
                     fill = 'skyblue') + 
    xlab('') + ylab('Frequency') + 
    ggtitle(phenotype)
}

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```


## **Genotype Data Quality Control**

Common techniques used to filter GWAS study data usually include removing **individuals** with more than 10% missing data across all genotypes, removing **genotypes** with more than 5% missing data across all individuals, and removing **genotypes** with an MAF less than 5%. No missing values were identified in the gentoype data, thus the first two points are satisfied. From the histogram below, we can see that no genotypes with an MAF of less than 5% are present in our data.

```{r, fig.height = 2}
MAFs <- 
  genotypes %>% 
  apply(2, function(x) {
    length(x[x > 0]) / length(x)
  })

MAFs %>% 
  data.frame(MAF = .) %>% 
  ggplot() + 
    geom_histogram(aes(x = MAF), 
                   color = 'black', 
                   fill = 'skyblue') + 
    geom_vline(aes(xintercept = 0.05, 
                   color = 'red_dotted_line'), 
               lty = 2)  + 
    xlab('Minor Allele Frequency') + 
    ylab('Frequency') + 
    scale_color_manual(values = c('red_dotted_line' = 'red'), 
                       labels = c('MAF = 5%')) + 
  theme(legend.title = element_blank())
```

### **Principle Components Analysis**

In order to understand if any other population structure exists in the data other than the captured population identifiers in the covariates, we now perform Principle Components Analysis (PCA) on the genotype data of each population seperately, and plot the population-seperated data by the two most discriptivive principle components.

Below we see that **CEU** and **TSI** populations are fairly uniform, and slight seperation is seen in both **FIN** and **GBR**. Both of these observed sub-populations do not seem to be extreme or due to the sex of the individual (indicated by the color of the points).

```{r}
unique_populations <- 
  covars$Population %>% 
  unique()
  
pca_results <- vector('list', 4)

for (i in seq_along(unique_populations)) {
  population <- unique_populations[i]
  pca_results[[i]] <-
    prcomp(genotypes[covars$Population == population, ], 
           center = TRUE)
}
```

```{r}
plot_list <- vector('list', 5)
names(plot_list) <- unique_populations

for (i in seq_along(unique_populations)) {
  population <- unique_populations[i]
  
  plot_list[[i]] <- 
    data.frame(PC1 = pca_results[[i]]$x[, 1], 
               PC2 = pca_results[[i]]$x[, 2], 
               Sex = covars[covars$Population == population, 'Sex']) %>% 
    ggplot() + 
      geom_point(aes(x = PC1, y = PC2, 
                     color = Sex)) +
    ggtitle(population) + 
    theme(legend.position = 'none')
}

(plot_list[[1]] + plot_list[[2]]) /
  (plot_list[[3]] + plot_list[[4]])

```

## **Performing GWAS Without Covariates**

We will first perfrom the five GWAS analyses **without** any included covariates, thus the linear models we will be fitting are of the form $Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + \epsilon$
where $\epsilon ~ N(0, \sigma^2)$. The models were fit using the `lm()` function in the R programming language, followed by calculation of the p-value through the models F-statistic on a distribution with the appropriate degrees of freedom.


```{r}
perform_gwas_no_covs <- function(genotypes, phenotype) { 
  n_genotypes <- ncol(genotypes)
  
  gwas_results <- 
    mclapply(seq_len(n_genotypes), 
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] + 
                                Xd[,genotype])
        
        p_val <-
          summary(model) %>% 
          .$fstatistic %>% 
          {pf(.[1], .[2], .[3], 
              lower.tail = FALSE)} 
      
        names(p_val) <- 
          colnames(genotypes)[genotype]
        
        p_val
      }, mc.cores = 7)
  
  unlist(gwas_results)
}
```

```{r}
gwas_no_covs_results <- 
  lapply(seq_len(ncol(phenotypes)), 
        function(i) {
          perform_gwas_no_covs(genotypes, 
                               phenotypes[,i])
   }) %>% 
  set_names(colnames(phenotypes))
```

### **QQ Plots**

In order to observe if the resultant ditribution of p-values for each GWAS is approximately normal, as we would expect under $H_0$, we produce a Quantile-Quantile plot comparing the quantiles of the p-value distribution with that of a uniform distribution.


```{r}
plot_uniform_qq <- function(p_values, pheno_name) {
  
  uniform_quantiles <- 
    qunif(ppoints(length(p_values)))
  
  p <- 
    data.frame(p_values          = sort(p_values),
               uniform_quantiles = sort(uniform_quantiles)) %>% 
    ggplot() + 
      geom_point(aes(x = -log10(uniform_quantiles),
                     y = -log10(p_values))) +
      geom_abline(intercept = 0, 
                  slope = 1, 
                  color="red", 
                  lty = 2) + 
      xlab('-log10(Uniform Quantiles)') +
      ylab('-log10(P Values)') + 
      ggtitle(pheno_name)
    
  p
}
```

```{r, fig.height = 3}
phenotype_names <- names(gwas_no_covs_results)

plot_list <- 
  lapply(seq_along(gwas_no_covs_results), 
        function(i) {
          plot_uniform_qq(gwas_no_covs_results[[i]], 
                          phenotype_names[[i]])
        })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```

### **Manhattan Plots**

To illistate the location and significance of the detected hits in the analyses perform, we plot the $-log_{10}(pvalue)$ of each genotype at the position where the genotype is found in the human genome.


```{r}
plot_manhattan <- function(p_values, plot_title, 
                           start = NULL, end = NULL) {
  

  ## Correlate p-values to genomic locations with snp_info
  unique_chromosomes <- 
    snp_info$chromosome %>% 
    unique()
  
  chromosome_offsets <- 
    sapply(unique_chromosomes, 
      function(chromosome) {
        offset <- 
          max(snp_info[snp_info['chromosome'] == chromosome,
                       'position'])
        names(offset) <- chromosome
        offset
    })
  
  genomic_locations <-
    snp_info[snp_info['id'] == names(p_values), ]  %>% 
    mutate(position = 
             if_else(chromosome == 1, 
                     position, 
                     position + sum(chromosome_offsets[1:chromosome-1]))) %>% 
    .$position
  
  n_genotypes <- length(p_values)
  
  df <- 
    data.frame(
      genome_location = genomic_locations, 
      pval = p_values) %>% 
    mutate(plot_pval = -log10(pval))
  
  # localizing manhattan plot to better show 
  # hits in the provided range
  if(!is.null(start) && !is.null(end)) {
    df <- df %>% 
      filter(genome_location >= start &
             genome_location <= end)
  } 
  
  
  p <- 
    df %>% 
    ggplot() + 
    geom_point(aes(x = genome_location, y = plot_pval), 
               col = 'skyblue', shape = 21) +
    geom_hline(aes(yintercept = -log10(0.05 / n_genotypes),
               color = 'red'), lty = 2, lwd = 1.1) +
    xlab('Genomic Location') + ylab('-log10(p value)') +
    ggtitle(plot_title) + ylim(c(0, 90)) + 
    scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) + 
    theme(legend.position = 'none', 
          axis.text.x = element_text(angle = -30)) 
  
  p
}
```


```{r, fig.height = 3.5}
plot_list <- 
  lapply(seq_along(gwas_no_covs_results), 
      function(i) {
        gwas_result <- gwas_no_covs_results[[i]]
        plot_name <- names(gwas_no_covs_results)[[i]]
        plot_manhattan(gwas_result, plot_name)
    })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])

```

Of the three phenotypes that we have observed significant hits for, we will take a closer look at the local regions where the hits were found to be.

```{r, fig.height = 2.5}
plot_manhattan(gwas_no_covs_results[['ERAP2']], 
               'ERAP2', 
               start = 9.65e7, 
               end = 9.725e7) + 
plot_manhattan(gwas_no_covs_results[['PEX6']],
               'PEX6',
               start = 4.27e7,
               end = 4.325e7) +
plot_manhattan(gwas_no_covs_results[['FAHD1']],
               'FAHD1',
               start = 1.35e6,
               end = 2.2e6)
```


## **Performing GWAS With Covariates**

We now perfrom the same five GWAS analyses **with** the inclusion of the collected **Population** and **Sex** covariates, thus the linear models we will be fitting are of the form $Y = \beta_{\mu} + X_a \beta_a + X_d \beta_d + X_{Sex} \beta_{Sex} + X_{Population} \beta_{Population} + \epsilon$
where $\epsilon ~ N(0, \sigma^2)$. The models were fit similarly using the `lm()` function in the R programming language, followed by calculation of the p-value through the models F-statistic on a distribution with the appropriate degrees of freedom.

```{r}
perform_gwas_with_covs <- function(genotypes, phenotype) { 
  n_genotypes <- ncol(genotypes)
  
  gwas_results <- 
    mclapply(seq_len(n_genotypes), 
      function(genotype) {
        model <- lm(phenotype ~ Xa[,genotype] + 
                                Xd[,genotype] + 
                                covars$Sex + 
                                covars$Population)
        
        p_val <-
          summary(model) %>% 
          .$fstatistic %>% 
          {pf(.[1], .[2], .[3], 
              lower.tail = FALSE)} 
      
        names(p_val) <- 
          colnames(genotypes)[genotype]
        
        p_val
      }, mc.cores = 7)
  
  unlist(gwas_results)
}
```

```{r}
gwas_with_covs_results <- 
  lapply(seq_len(ncol(phenotypes)), 
        function(i) perform_gwas_with_covs(genotypes, phenotypes[,i])) %>% 
  set_names(colnames(phenotypes))
```


### **QQ Plots**

In order to observe if the resultant ditribution of p-values for each GWAS is approximately normal, as we would expect under $H_0$, we produce a Quantile-Quantile plot comparing the quantiles of the p-value distribution with that of a uniform distribution.

```{r, fig.height = 3}
phenotype_names <- names(gwas_with_covs_results)

plot_list <- 
  lapply(seq_along(gwas_with_covs_results), 
        function(i) {
          plot_uniform_qq(gwas_with_covs_results[[i]], 
                          phenotype_names[[i]])
        })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])
```

### **Manhattain Plots**

To illistate the location and significance of the detected hits in the analyses perform, we plot the $-log_{10}(pvalue)$ of each genotype at the position where the genotype is found in the human genome.

```{r, fig.height = 3.5}
plot_list <- 
  lapply(seq_along(gwas_with_covs_results), 
      function(i) {
        gwas_result <- gwas_with_covs_results[[i]]
        plot_name <- names(gwas_with_covs_results)[[i]]
        plot_manhattan(gwas_result, plot_name)
    })

(plot_list[[1]] + plot_list[[2]]) / 
  (plot_list[[3]] + plot_list[[4]] + plot_list[[5]])

```

Of the three phenotypes that we have observed significant hits for, we will take a closer look at the local regions where the hits were found to be. We see below that we have found the same hits with and without covariates included in our model.

```{r, fig.height = 2.5}
plot_manhattan(gwas_with_covs_results[['ERAP2']], 
               'ERAP2', 
               start = 9.65e7, 
               end = 9.725e7) + 
plot_manhattan(gwas_with_covs_results[['PEX6']],
               'PEX6',
               start = 4.27e7,
               end = 4.325e7) +
plot_manhattan(gwas_with_covs_results[['FAHD1']],
               'FAHD1',
               start = 1.35e6,
               end = 2.2e6)
```

## **Table of Most Significant Hits**

```{r}
top_sig_hits_no_covs <- vector('list', 5)
top_sig_hits_with_covs <- vector('list', 5)
num_sig_no_covs <- vector('numeric', 5)
num_sig_with_covs <- vector('numeric', 5)

names(top_sig_hits_no_covs) <- phenotype_names
names(top_sig_hits_no_covs) <- phenotype_names
names(num_sig_no_covs) <- phenotype_names
names(num_sig_with_covs) <- phenotype_names

for (phenotype in phenotype_names) {
  num_sig_no_covs[phenotype] <- 
    gwas_no_covs_results[[phenotype]] %>% 
    {. * length(.)} %>% {. < 0.05} %>% which() %>% length()
  
  num_sig_with_covs[phenotype] <- 
    gwas_with_covs_results[[phenotype]] %>% 
    {. * length(.)} %>% {. < 0.05} %>% which() %>% length()
  
  top_sig_hits_no_covs[[phenotype]] <-
    gwas_no_covs_results[[phenotype]] %>% 
    sort() %>% head(3) %>% names() %>% paste0(collapse = ',')
  
  top_sig_hits_with_covs[[phenotype]] <-
    gwas_with_covs_results[[phenotype]] %>% 
    sort() %>% head(3) %>% names() %>% paste0(collapse = ',')
}
```


```{r, eval = FALSE}
data.frame(
  Phenotype = c('ERAP2', 'PEX6', 'FAHD1'), 
  'Num Sig. without covs' = c(num_sig_no_covs['ERAP2'], 
                                    num_sig_no_covs['PEX6'],
                                    num_sig_no_covs['FAHD1']), 
  'Num Sig. with covs' =  c(num_sig_with_covs['ERAP2'], 
                                    num_sig_with_covs['PEX6'],
                                    num_sig_with_covs['FAHD1']), 
  'Top 3 Sig. without covs' = c(top_sig_hits_no_covs[['ERAP2']], 
                                      top_sig_hits_no_covs[['PEX6']], 
                                      top_sig_hits_no_covs[['FAHD1']]), 
  'Top 3 Sig. with' = c(top_sig_hits_with_covs[['ERAP2']], 
                                     top_sig_hits_with_covs[['PEX6']], 
                                     top_sig_hits_with_covs[['FAHD1']])
  ) %>% 
  kable()
```


Analysis | ERAP2 # of Significant Genotypes | ERAP2 3 Most Significant Genotypes | PEX6 # of Significant Genotypes | PEX6 3 Most Significant Genotypes | FAHD1 # of Significant Genotypes | FAHD1 3 Most Significant Genotypes
--- | --- | --- | --- | --- | --- | ---
Without Covariates | 73 | rs7726445, rs7731592, rs2161548 | 29 | rs1129187, rs10948061, rs9471985 | 90 | rs11644748, rs140254902, rs9652776
--- | --- | --- | --- | --- | --- | ---
With One Hot Encoded Covariates | 73 | rs7726445, rs7731592, rs2161548 | 27 | rs1129187, rs10948061, rs9471985 | 90 | rs11644748, rs140254902, rs9652776
--- | --- | --- | --- | --- | --- | ---
With Linearly Encoded Covariates | 74 | rs7726445, rs7731592, rs2161548 | 28 | rs1129187, rs10948061, rs9471985 | 90 | rs11644748, rs140254902, rs9652776


## **Biological Support of Found Hits**

**TODO**: Maybe just look up Genecards for the genes that corespond to the most significant hits?

Using the Manhattan plots above, we can search through published research online for SNPs with known biologicalfunctions that are located where our GWAS identified a locus for a causal polymorphism. Since we have onlyreceived a sample of the SNPs in the genome, we may have found the location of a specific genotype that isnot included in our small sample. Therefore, we search for known causal genotypes determined by eQTLexperiments that are located in the regions where there are many hits in the ERAP2, PEX6, and FAHD1genes.7

For ERAP2, the eSNP rs2927608 has been previously discovered as a causal genotype **[1]**. This SNP is locatedat position 96916728, which is in the region where many of our significant genotypes lie for ERAP2, and isassociated with inflammatory bowel disease through an increased gene expression of ERAP2. The genotypeof this locus alone accounted for 74.5% of the total variability of ERAP2 expression in blood in this study.

For PEX6, the eSNP rs10948059 is located at position 42960723 in the genome. This corresponds to thelocus where our significant genotypes were found using our GWAS analysis. This SNP has been associatedwith prostate cancer from previous association studies **[2]** . The results of the literature imply that variantsassociated with prostate cancer can be identified through expressional change in the PEX6 gene.

For FAHD1, the eSNP rs1065656 is located at position 1788835 in the genome. We have found significantgenotypes at this location in our GWAS analysis. Literature has demonstrated that rs1065656 is associatedwith circulating IGF‐I and IGFBP‐3 concentration, which have been implicated in risk of human diseasesincluding cardiovascular diseases, diabetes, and cancer **[3]**.


## **References**

Di Narzo, A., Peters, L., Argmann, C., Stojmirovic, A., Perrigoue, J., Li, K., . . . Hao, K. (2016, June23). Blood and Intestine Eqtls from AN Anti-TNF-Resistant Crohn’s DISEASE Cohort inform IBDgenetic association loci. Retrieved May 04, 2021, from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4931595/#sup12.

**[2]** Lee, J., Ryu, J., & Lee, C. (2016, December). Strong cis-acting expression quantitative trait loci for thegenes encoding snhg5 and pex6. Retrieved May 04, 2021, from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5207599/

[3] Teumer, A., Qi, Q., Nethander, M., Aschard, H., Bandinelli, S., Beekman, M., . . . Kaplan, R. (2016,October). Genomewide meta-analysis IDENTIFIES loci associated WITH IGF-I and IGFBP-3 levelswith impact on age-related traits. Retrieved May 04, 2021, from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5013013
